pkg/sql/create_table.go

```diff
@@ -187,6 +187,14 @@ func (n *createTableNode) startExec(params runParams) error {
 		telemetry.Inc(sqltelemetry.CreateTempTableCounter)
 	}
 
+	// TODO(#46556): support ON COMMIT DROP | DELETE ROWS
+	if !isTemporary && n.n.OnCommit != tree.CreateTableOnCommitUnset {
```

Comment:
`return unimplemented.NewWithIssue(46556, ....)`

Reply:
This is never intended to be implemented - it is an error as `ON COMMIT` only works on temporary table.

This matches Postgres.

Reply:
what is the TODO about then?

Reply:
let me expand on it :)